python/hail/docs/hailpedia/expressions.rst

```diff
@@ -0,0 +1,328 @@
+For example, we might want to return ``1`` if ``x < -1``, ``2`` if
+``-1 <= x <= 2`` and ``3`` if ``x > 2``.
+
+    >>> (hl.case()
+    ...   .when(x < -1, 1)
+    ...   .when((x >= -1) & (x <= 2), 2)
+    ...   .when(x > 2, 3)
+    ...   .or_missing())
+    <Int32Expression of type int32>
+
+Notice that this expression ends with a call to :meth:`.CaseBuilder.or_missing`,
+which means that if none of the conditions are met, a missing value is returned.
+
+Cases started with :func:`.case` can end with a call to
+:meth:`.CaseBuilder.or_missing`, :meth:`.CaseBuilder.default`, or
+:meth:`.CaseBuilder.or_error`, depending on what you want to happen if none
+of the *when* clauses are met.
+
+It's important to note that missingness propagates up in Hail, so if the value
+of the discriminant in a case statement is missing, then the result will be
+missing as well.
+
+>>> y = hl.null(hl.tint32)
+>>> result = hl.case().when(y > 0, 1).default(-1)
+>>> result.value
+
+The value of ``result`` will be missing, not ``1`` or ``-1``, because the
+discriminant, ``y``, is missing.
+
+Switch Statements
+~~~~~~~~~~~~~~~~~
+
+Finally, Hail has the :func:`.switch` function to build a conditional tree based
+on the value of an expression. In the example below, ``csq`` is a
+:class:`.StringExpression` representing the functional consequence of a
+mutation. If ``csq`` does not match one of the cases specified by
+:meth:`.SwitchBuilder.when`, it is set to missing with
+:meth:`.SwitchBuilder.or_missing`. Other switch statements are documented in the
+:class:`.SwitchBuilder` class.
+
+    >>> csq = hl.str('nonsense')
+
+    >>> (hl.switch(csq)
+    ...    .when("synonymous", False)
+    ...    .when("intron", False)
+    ...    .when("nonsense", True)
+    ...    .when("indel", True)
+    ...    .or_missing())
+    <BooleanExpression of type bool>
+
+As with case statements, missingness will propagate up through a switch
+statement. If we changed the value of ``csq`` to the missing value
+``hl.null(hl.tstr)``, then the result of the switch statement above would also
+be missing.
+
+Missingness
+===========
+
+In Hail, all expressions can be missing. An expression representing a missing
+value of a given type can be generated with the :func:`.null` function, which
+takes the type as its single argument.
+
+An example of generating a :class:`.Float64Expression` that is missing is:
+
+    >>> hl.null('float64')
+    <Float64Expression of type float64>
+
+These can be used with conditional statements to set values to missing if they
+don't satisfy a condition:
+
+    >>> hl.cond(x > 2.0, x, hl.null(hl.tfloat))
+    <Float64Expression of type float64>
+
+The Python representation of a missing value is ``None``. For example, if
+we define ``cnull`` to be a missing value with type :class:`.tcall`, calling
+the method `is_het` will return ``None`` and not ``False``.
+
+    >>> cnull = hl.null('call')
+    >>> cnull.is_het().value
+    None
+
+
+Binding Variables
```

Comment:
I removed this section in #4104 since it's no longer accurate. Not sure how the merge will happen here since it's moved to a different file, but can you make sure that this gets removed in this PR as well?

Reply:
will do!